experimenter/experimenter/nimbus_ui/templatetags/nimbus_extras.py

```diff
@@ -97,3 +99,53 @@ def can_review_experiment(context, experiment):
 @register.filter
 def qa_icon_info(value):
     return NimbusConstants.QAStatus.get_icon_info(value)
+
+
+@register.filter
+def channel_icon_info(value):
+    return NimbusExperiment.Channel.get_icon_info(value)
+
+
+@register.simple_tag
+def render_channel_icons(experiment):
+    channels_data = []
+
+    if experiment.is_desktop and experiment.channels:
+        for channel in sorted(experiment.channels):
+            icon_info = NimbusExperiment.Channel.get_icon_info(channel)
+            channel_label = experiment.Channel(channel).label
+            channels_data.append(
+                {
+                    "icon_info": icon_info,
+                    "label": channel_label,
+                    "is_multi": True,
+                }
+            )
+    elif experiment.channel:
+        icon_info = NimbusExperiment.Channel.get_icon_info(experiment.channel)
+        channel_label = experiment.Channel(experiment.channel).label
+        channels_data.append(
+            {
+                "icon_info": icon_info,
+                "label": channel_label,
+                "is_multi": False,
+            }
+        )
+
+    return channels_data
```

Comment:
Yeah exactly 🙏 

Reads very cleanly here, keeps the html in the templates.  Very nice 🎉 🎉 🎉 🎉